modules/core/src/main/java/org/apache/synapse/rest/RESTRequestHandler.java

```diff
@@ -158,6 +170,7 @@ private boolean identifyAPI(API api, MessageContext synCtx, List defaultStrategy
             if (log.isDebugEnabled()) {
                 log.debug("Located specific API: " + api.getName() + " for processing message");
             }
+            synCtx.setProperty(RESTConstants.PROCESSED_API, api);
```

Comment:
shall we move this to process() method of the API so that we don't need to set it over and over again here into different cases?

Reply:
done